src/commons/utils/alerts.py

```diff
@@ -0,0 +1,168 @@
+        :param receiver: receiver of mail
+        :param tp_id : Test Plan ID to be sent in subject.
+        :param report_path: Report file path.
+        """
+        super().__init__()
+        self.event_fail = threading.Event()
+        self.mail_obj = Mail(sender=sender, receiver=receiver)
+        self.sender = sender
+        self.receiver = receiver
+        self.tp_id = tp_id
+        self.health_obj = get_logical_node()
+        self.report_path = report_path
+        self.message_id = None
+
+    def prepare_email(self, execution_status) -> MIMEMultipart:
+        """
+        Prepare email message with format and attachment
+        :param execution_status: Execution status. In Progress/Fail
+        :return: Formatted MIME message
+        """
+        hctl_status = json.dumps(self.health_obj.get_hctl_status()[1], indent=4)
+        result, pod_status = self.health_obj.execute_command(commands.CMD_POD_STATUS)
+        subject = f"TestPlan {self.tp_id} {execution_status} on {self.health_obj.host}"
+        body = f"""<h3>TestPlan {self.tp_id} {execution_status} on {self.health_obj.host}.</h2>\n
+                   <h3>PFA hctl cluster status, pod status & execution status.</h3>\n"""
+        build_url = os.getenv("BUILD_URL")
+        if build_url:
+            body += f"""Visit Jenkins Job: <a href="{build_url}">{build_url}</a>"""
+        message = MIMEMultipart()
+        message['From'] = self.sender
+        message['To'] = COMMASPACE.join(self.receiver.split(','))
+        message['Date'] = formatdate(localtime=True)
+        message['Subject'] = subject
+        if not self.message_id:
+            self.message_id = make_msgid()
+            message["Message-ID"] = self.message_id
+        else:
+            message["In-Reply-To"] = self.message_id
+            message["References"] = self.message_id
+
+        attachment = MIMEApplication(hctl_status, Name="hctl_status.txt")
+        attachment['Content-Disposition'] = 'attachment; filename=hctl_status.txt'
+        message.attach(attachment)
+
+        if result:
+            attachment = MIMEApplication(pod_status, Name="pod_status.txt")
+            attachment['Content-Disposition'] = 'attachment; filename=pod_status.txt'
+            message.attach(attachment)
+        else:
+            body += """<h3>Could not collect pod status</h3>"""
+        name = os.path.basename(self.report_path)
+        if os.path.exists(self.report_path):
+            message.attach(MIMEText(body, "html"))
+            with open(self.report_path, "rb") as fil:
+                attachment = MIMEApplication(fil.read(), Name=name)
+            attachment['Content-Disposition'] = f'attachment; filename={name}'
+            message.attach(attachment)
+        else:
+            message.attach(MIMEText(body + f"<h3>Could not find {self.report_path}.</h3>", "html"))
+        return message
+
+    def run(self):
+        """
+        Send Mail notification periodically.
```

Comment:
Docstring should be single liner, if fit within line limit.

Reply:
Done